disk_asm.c

```
@@ -101,27 +120,47 @@ int disk_asm_read_cluster(int clu_index, unsigned char *buf) {
+    mem_domain_free(MEM_DOMAIN_FS, tmp);
+    free(line);
     return 0;
 }
 
 /* Write raw bytes to cluster (uses ASM for buffer prep) */
 int disk_asm_write_cluster(int clu_index, const unsigned char *buf) {
     if (clu_index < 0 || clu_index >= g_total_clusters)
         return -1;
+    if (g_disk_host_fat32) {
+        uint64_t off = 0;
+        if (fat32_host_shell_cluster_byte_offset(clu_index, &off) != 0)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)